client/src/templates/Challenges/classic/Editor.js

```diff
@@ -134,6 +140,20 @@ class Editor extends Component {
         this.props.setEditorFocusability(false);
       }
     });
+    this._editor.addAction({
+      id: 'save-editor-content',
+      label: 'Save editor content to localStorage',
+      keybindings: [
+        monaco.KeyMod.chord(monaco.KeyMod.CtrlCmd | monaco.KeyCode.KEY_S)
+      ],
+      run: () => {
+        this.props.saveEditorContent();
+        this.props.createFlashMessage({
```

Comment:
Could you move the message creation into the code storage epic?  Reason being, we only want to display the message if it succeeds.  Also, ideally, a warning message should be shown if it fails.

Reply:
Yeah! That makes a lot of sense.

Reply:
Thanks for moving the code.  However, it still shows the success flash regardless of what happens to the store.  The logic should be:

```
if (new files != stored files) {
  // store new files
  if (new files != stored files) {
    // report that they couldn't be saved
  } else {
    // report that they have been saved
  }
} else {
  // move on, since the user hasn't changed anything
}

```

Give me a shout if that's unclear/tricky to implement.

Reply:
I think this could be done using a try-catch block.  Heres what I found about handing localStorage errors. http://chrisberkhout.com/blog/localstorage-errors/

```
try {
  store.set(id, files);
  // report code was saved
} catch (e) {
  // report code was not saved
}
```

Would this work?

Reply:
I completely missed this comment, sorry!  

From what I remember, if you disable local storage, it fails quietly (i.e. doesn't store anything and doesn't throw an error).  Good point about the try-catch block - I think is necessary and it probably needs following by the check I outlined above.

Reply:
Okay sounds good.  I see we have not heard back yet about whether or not this should be 2 separate pull requests.  I think that you are correct though. I'm considering just going ahead with that idea, closing this pull request, and reopening one that just deals with saving the code; no flash message.  I don't see how not having the flash message initially would be a big issue.  It may not being the greatest UX at first, but most of the code to implement the flash message is here so it shouldn't take long to get that PR through as well.  What do you think?  Should we wait to hear back?

Reply:
What makes most sense to me is this: one PR for code saving (including a flash message) and second PR that just addresses the flash message UI.

So, I'd keep this PR focused on a) storing the code and b) creating an appropriate flash message in response to that.  Anything that changes the message UI should go in the second PR.

> Okay sounds good. I see we have not heard back yet about whether or not this should be 2 separate pull requests.

True, I was more floating the idea last time we talked about this.  For the sake of moving things forward, I'll make a decision: please split this into two PRs as discussed above.  I'd keep this PR open (just remove stuff that changes how messages display), but if closing this and creating two fresh ones is easier, that works too!

Reply:
Hey sorry for the late response, I'm working on this now.  I just want to clarify something before I make these changes.   If I remove the changes to the flash message file (and the corresponding css files I changed) that means that the flash message will overlay content on the coding screen.  The fix to that will be in the other PR that deals with changes to the flash message.  Is that going to be to alright?

Reply:
@jakethecoder95, yes that should be fine.  There's a [code freeze](https://github.com/freeCodeCamp/freeCodeCamp/issues/38308) at the moment, so there might be a little delay before this can be merged.   It should not be be for too long, though.

Reply:
@ojeytonwilliams  Thanks for you patience.  I had to move this to two new PRs cause I made the original changes in my master branch and needed to put them onto their own branches.  

Here is the new changes:
Storage PR: #38324 
Flash Message: #38327